modin/test/storage_formats/omnisci/test_internals.py

```diff
@@ -0,0 +1,88 @@
+# Licensed to Modin Development Team under one or more contributor license agreements.
+# See the NOTICE file distributed with this work for additional information regarding
+# copyright ownership.  The Modin Development Team licenses this file to you under the
+# Apache License, Version 2.0 (the "License"); you may not use this file except in
+# compliance with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+import pytest
+import sys
+import subprocess
+
+
+@pytest.mark.parametrize(
+    "import_strategy",
+    [
+        pytest.param(
+            """
+import modin.config as cfg
+cfg.Engine.put('Native') # 'omniscidbe'/'dbe' would be imported with dlopen flags first time
+cfg.StorageFormat.put('Omnisci')
+cfg.IsExperimental.put(True)
+import modin.pandas as pd
```

Comment:
This should be working on current master. We need to check importing omnisci and pyarrow.gandiva, but not Modin.
```suggestion
import modin.config as cfg
cfg.Engine.put('Native') # 'omniscidbe'/'dbe' would be imported with dlopen flags first time
cfg.StorageFormat.put('Omnisci')
cfg.IsExperimental.put(True)
import pyarrow.gandiva as gandiva
```

Reply:
As far as I understand, `PyDBEngine` and `pyarrow.gandiva` packages are incompatible [because LLVM issue](https://github.com/modin-project/modin/pull/3783#discussion_r767887619) that we can't and shouldn't fix in Modin, we just don't want to import obviously incompatible packages in the natural Modin's initialization flow. And I believe that's the thing we want to test here.

Importing OmniSci + pyarrow gandiva would always fail because of non dependent on Modin reasons, so why should we test it here?

Reply:
Checking `PyDbEngine` and `pyarrow.gandiva` imports we could potentially find out a point when the packages become compatible (`pyarrow.gandiva` or we change something in building the package).
Also, we can add a test like this
```python
import modin.config as cfg
cfg.Engine.put('Native') # 'omniscidbe'/'dbe' would be imported with dlopen flags first time
cfg.StorageFormat.put('Omnisci')
cfg.IsExperimental.put(True)
import modin.core.storage_formats
```
so when we move importing pyarrow.gandiva at the header of the PyarrowQC, we will get this test failing and that will be a point to remove the test and make `storage_formats` module independent from a concrete storage format.

Reply:
> Checking PyDbEngine and pyarrow.gandiva imports we could potentially find out a point when the packages become compatible ... and that will be a point to remove the test and make storage_formats module independent from a concrete storage format.

The idea of clearing [`storage_formats/__init__.py`](https://github.com/modin-project/modin/blob/cf1e5416a1ba4b072d3719bcfa94b8dbaddfdbb4/modin/core/storage_formats/__init__.py) and then reverting it back when `PyDbEngine` and `pyarrow.gandiva` would become compatible feels wrong to me. We actually already faced the similar problem of importing too much unneeded in the `storage_formats/__init__.py` at #2126. There we fixed the problem of unnecessary `pyarrow` import by [wrapping it into `try-except` construction](https://github.com/modin-project/modin/blob/cf1e5416a1ba4b072d3719bcfa94b8dbaddfdbb4/modin/core/storage_formats/__init__.py#L20-L25) with a note, that we can do a [lazy submodules import after we bump the minimum python version to 3.7+](https://github.com/modin-project/modin/pull/2126#discussion_r493394176), which we actually did about a year ago (#2615). So now, I think it's time to implement the mentioned lazy submodules import instead of trying to joggle with incompatible packages.

So I don't see a reason why we should make a test checking that `PyDBEngine` and `pyarrow.gandiva` are incompatible since it's not Modin's care. The only thing that is logical to me is to test that the natural Modin initialization flow works properly witn OmniSci.

Reply:
There is probably misunderstanding between us, let's discuss offline.

Reply:
As the result of the offline discussion, agreed on:
1. Leave the existing test as they are.
2. Add a new test explicitly importing PyDbEngine + pyarrow.ganvida, that would indicate the current status of compatibility of these two packages.